src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -823,6 +828,29 @@ private static void TryGetAssemblyNameComponent(string fusionName, string compon
             }
         }
 
+        /// <summary>
+        /// Tries to create an <see cref="AssemblyNameExtension"/> out of a primary reference metadata.
+        /// </summary>
+        private static AssemblyNameExtension GetAssemblyNameFromItemMetadata(ITaskItem item)
+        {
+            string version = item.GetMetadata(ItemMetadataNames.assemblyVersion);
+            if (string.IsNullOrEmpty(version))
+            {
+                return null;
+            }
+
+            string publicKeyToken = item.GetMetadata(ItemMetadataNames.publicKeyToken);
+            if (string.IsNullOrEmpty(publicKeyToken))
+            {
+                return null;
+            }
+
+            string name = Path.GetFileNameWithoutExtension(item.ItemSpec);
```

Comment:
Think we should have a check for if ItemSpec is empty? (Can ItemSpec be empty?)

Reply:
The code has been updated to obtain the name from metadata.